### PR TITLE
Feature/entity linking

### PR DESCRIPTION
**Pull request by geoffreyvd**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,5 +17,10 @@ RUN python3 -m spacy download en_core_web_md
 
 RUN CFLAGS="-Wno-narrowing" pip3 install -r requirements.txt
 
+<<<<<<< HEAD
+# # Install spacy data
+# RUN python3 -m spacy download en_core_web_md
+=======
```
Comment (kevinh1998) [inline Dockerfile]: Dit hele gedeelte kan weg

**Review by kevinh1998 — CHANGES_REQUESTED** [review]
1 dingetje voor de rest goed!

File: requirements.txt [hunk 2]
```diff
@@ -14,7 +14,6 @@ click==7.1.2
 cryptography==2.6.1
 cymem==2.0.4
 elasticsearch==7.9.1
-en-core-web-md==2.3.1
```
Comment (kevinh1998) [inline requirements.txt:17]: Deze er ook in laten trouwens

File: src/entity_linking.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    label = hit["_source"]["schema_name"]
+                except Exception as e:
+                    continue
+                id = hit["_id"]
+                #could also retrieve the ES score here
+                id_labels.setdefault(id, set()).add(label)
+        return id_labels
+
+    def searchTrident(self,query):
+        query="PREFIX wde: <http://www.wikidata.org/entity/> "\
```
Comment (misharigot) [inline src/entity_linking.py]: The `query` arg is being overwritten immediately by this line.

File: src/entity_linking.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            "select ?s where { ?s wdp:P31 wde:Q515 . } LIMIT 10"
+        results = db.sparql(query)
+        json_results = json.loads(results)
+        return json_results["results"]
+
+    #!!a bit confusing but i use _ to anotate tuples, where the left side of _ means the first item in the tuple!!
+    def entityLinking(self, entitytype_entities):
+        entity_wikidata = [] #(entity, wikidata)
+        for entitytype_entity in entitytype_entities:
+            entitiy_popularity = [] #(entity, popularity)
```
Comment (misharigot) [inline src/entity_linking.py]:
```suggestion
            entity_popularity = [] #(entity, popularity)

```

File: requirements.txt (hunk 2 above)
Comment (geoffreyvd) [reply requirements.txt:17]: scherp

File: src/test_elasticsearch_server.py [hunk 5]
```diff
@@ -4,13 +4,17 @@
...
     response = e.search(index="wikidata_en", body=json.dumps(p))
     id_labels = {}
     if response:
         for hit in response["hits"]["hits"]:
-            label = hit["_source"]["schema_name"]
+            try:
+                #same entity have schema name missing
+                label = hit["_source"]["schema_name"]
+            except Exception as e:
+                continue
```
Comment (misharigot) [inline src/test_elasticsearch_server.py:17]: Wouldn't an exception be useful here, as the aim is to test ES?

Comment (geoffreyvd) [reply src/test_elasticsearch_server.py:17]: Debatable, depends in what way you want to test, if i want to query elasticsearch without it crashing all the time...

**Pull request merged**